File: src/models/trace.py
from pydantic import BaseModel,Field

class Span(BaseModel):
    id: str
    name: str
    start_time: float
    end_time: float
    duration: float
    spans: list['Span'] = []


class Trace(BaseModel):
    trace_id:str
    start_time: float
    end_time: float
    duration: float
    service_name: str |None = None
    spans: list[Span] = Field(default_factory=list)

    

File: ui/src/models/trace.ts
export interface Span {
  id: string;
  name: string;
  start_time: number;
  end_time: number;
  duration: number;
  num_debug_logs?: number;
  num_info_logs?: number;
  num_warning_logs?: number;
  num_error_logs?: number;
  num_critical_logs?: number;
  spans?: Span[];
  telemetry_sdk_language?: string;
}

export interface Trace {
  id: string;
  service_name?: string;
  service_environment?: string;
  duration: number;
  start_time: number;
  end_time: number;
  percentile: string;
  num_warning_logs?: number;
  num_error_logs?: number;
  num_critical_logs?: number;
  spans: Span[];
  telemetry_sdk_language: string[];
}

export interface TraceResponse {
  success: boolean;
  data: Trace[];
  error?: string;
  next_pagination_token?: string;
  has_more?: boolean;
}
